modules/effects/migrations/9_0_0/index.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  chain,
+  Rule,
+  SchematicContext,
+  Tree,
+} from '@angular-devkit/schematics';
+import {
+  commitChanges,
+  visitTSSourceFiles,
+} from '@ngrx/effects/schematics-core';
+import {
+  createReplaceChange,
+  ReplaceChange,
+} from '@ngrx/effects/schematics-core';
+import * as ts from 'typescript';
+
+const OLD_TEXT = 'resubscribeOnError';
+const NEW_TEXT = 'useEffectsErrorHandler';
+
+function renameErrorHandlerConfig(): Rule {
+  return (tree: Tree, ctx: SchematicContext) => {
+    visitTSSourceFiles(tree, sourceFile => {
+      const createChange = (node: ts.Node) =>
+        createReplaceChange(sourceFile, node, OLD_TEXT, NEW_TEXT);
+
+      const changes: ReplaceChange[] = [
+        ...findCreateEffectsConfigKey(sourceFile, createChange),
+        ...findEffectDecoratorConfigKey(sourceFile, createChange),
+      ];
+
+      commitChanges(tree, sourceFile.fileName, changes);
+
+      if (changes.length) {
+        ctx.logger.info(
+          `[@ngrx/effects] Updated Effects configuration, see the migration guide (https://ngrx.io/guide/migration/v9#effects) for more info`
+        );
+      }
+    });
+  };
+}
+
+function findCreateEffectsConfigKey(
+  sourceFile: ts.SourceFile,
+  createChange: (node: ts.Node) => ReplaceChange
+) {
+  const changes: ReplaceChange[] = [];
+  ts.forEachChild(sourceFile, node => findKey(node, changes));
+  return changes;
+
+  function findKey(node: ts.Node, changes: ReplaceChange[]) {
+    if (
+      ts.isIdentifier(node) &&
+      node.text === OLD_TEXT &&
+      isEffectDecorator(node.parent.parent.parent.parent)
```

Comment:
I'm worried that this (`parent.parent.parent`) might throw an error in some cases.
What about finding the `createEffect` first, and then looking for `resubscribeOnError`?

Reply:
Hm I worried about this too, so tested it. Interestingly node.parent is never undefined, I think it just becomes recursive at the top level somehow. I agree though that it would be less confusing to find the effect context first and then descend into children so I'll change that 👍 

It will be a little more complex I suspect as I will need to make sure not to descend into the effect declaration as that could plausibly contain identifiers that I shouldn't be changing.

Reply:
Ok refactor done, I actually went with a different strategy (visitor pattern) which ended up being a lot cleaner